src/main/java/javax/jmdns/impl/DNSRecord.java

```diff
@@ -150,11 +155,33 @@ public boolean isStale(long now) {
     }
 
     /**
+     * Check if the record is stale and whether the record should be refreshed over the network.
+     *
+     * @param now
+     *            update date
+     * @return <code>true</code> is the record is stale and should be refreshed, <code>false</code> otherwise.
+     */
+    public boolean isStaleAndShouldBeRefreshed(long now) {
+        return getExpirationTime(_isStaleAndShouldBeRefreshedPercentage) <= now;
+    }
+
+    /*
+    * Increment the percentage that determines whether a record needs to be refreshed.
+     */
+    void incrementRefreshPercentage() {
+        _isStaleAndShouldBeRefreshedPercentage += DNSConstants.STALE_REFRESH_INCREMENT;
+        if (_isStaleAndShouldBeRefreshedPercentage > 100) {
+            _isStaleAndShouldBeRefreshedPercentage = 100;
+        }
+    }
+
+    /**
      * Reset the TTL of a record. This avoids having to update the entire record in the cache.
      */
     void resetTTL(DNSRecord other) {
         _created = other._created;
         _ttl = other._ttl;
+        _isStaleAndShouldBeRefreshedPercentage = DNSConstants.STALE_REFRESH_STARTING_PERCENTAGE + _randomStaleRefreshOffset;
```

Comment:
> In principle this should not matter since it is a random value that is only used to spread the network load over time.

Right, but then you could have just omitted that line. As it stands now it's a NOP.

> I would actually prefer to change the signature of resetTTL(DNSRecord) to resetTTL(created, ttl)

That decision is up to the jmdns@openhab maintainers. I would probably deprecate `resetTTL(DNSRecord)` and add `resetTTL(long, int)`.